openbravo@por0957:~/workspace/src/retail-modules/pi-mobile$ hg diff -c 7011
diff -r 51ce78428164 -r 27cf9832ab7a src-test/org/openbravo/test/mobile/retail/extmodules/selenium/tests/stockvalidation/I34089_OrderFromQuotationAllowNegativeStock.java
--- a/src-test/org/openbravo/test/mobile/retail/extmodules/selenium/tests/stockvalidation/I34089_OrderFromQuotationAllowNegativeStock.java	Thu Dec 21 17:02:55 2017 +0100
+++ b/src-test/org/openbravo/test/mobile/retail/extmodules/selenium/tests/stockvalidation/I34089_OrderFromQuotationAllowNegativeStock.java	Fri Dec 22 12:00:20 2017 +0100
@@ -60,9 +60,9 @@
 
     addProduct();
     verify(TestIdExtModules.LABEL_RECEIPT_ROW1_QTY, "1");
-    verify(TestIdExtModules.LABEL_RECEIPT_ROW2_QTY, "100");
-    verify(TestIdExtModules.LABEL_RECEIPT_TOTAL_QTY, "101");
-    verify(TestIdExtModules.LABEL_TOTALTOPAY, "15,053.60");
+    verify(TestIdExtModules.LABEL_RECEIPT_ROW2_QTY, "600");
+    verify(TestIdExtModules.LABEL_RECEIPT_TOTAL_QTY, "601");
+    verify(TestIdExtModules.LABEL_TOTALTOPAY, "90,303.60");
 
     tap(TestIdExtModules.BUTTON_PAY);
     verify(TestIdExtModules.LABEL_RECEIPT_TYPE, "Quotation - Under Evaluation");
@@ -94,11 +94,11 @@
     verify(TestIdExtModules.LABEL_TOTALTOPAY, "154.10");
     // changes quantity
     tap(TestIdPack.BUTTON_RECEIPT_ROW2);
-    tap(TestIdPack.BUTTON_KEYPAD_1);
+    tap(TestIdPack.BUTTON_KEYPAD_6);
     tap(TestIdPack.BUTTON_KEYPAD_0);
     tap(TestIdPack.BUTTON_KEYPAD_0);
     tap(TestIdPack.BUTTON_KEYPAD_QUANTITY);
-    verify(TestIdExtModules.LABEL_TOTALTOPAY, "15,053.60");
+    verify(TestIdExtModules.LABEL_TOTALTOPAY, "90,303.60");
   }
 
   private void selectQuantity(final String orgId, final String prodId) {
diff -r 51ce78428164 -r 27cf9832ab7a src-test/org/openbravo/test/mobile/retail/extmodules/selenium/tests/stockvalidation/I34089_OrderFromQuotationNotAllowNegativeStock.java
--- a/src-test/org/openbravo/test/mobile/retail/extmodules/selenium/tests/stockvalidation/I34089_OrderFromQuotationNotAllowNegativeStock.java	Thu Dec 21 17:02:55 2017 +0100
+++ b/src-test/org/openbravo/test/mobile/retail/extmodules/selenium/tests/stockvalidation/I34089_OrderFromQuotationNotAllowNegativeStock.java	Fri Dec 22 12:00:20 2017 +0100
@@ -48,9 +48,9 @@
 
     addProduct();
     verify(TestIdExtModules.LABEL_RECEIPT_ROW1_QTY, "1");
-    verify(TestIdExtModules.LABEL_RECEIPT_ROW2_QTY, "100");
-    verify(TestIdExtModules.LABEL_RECEIPT_TOTAL_QTY, "101");
-    verify(TestIdExtModules.LABEL_TOTALTOPAY, "15,053.60");
+    verify(TestIdExtModules.LABEL_RECEIPT_ROW2_QTY, "600");
+    verify(TestIdExtModules.LABEL_RECEIPT_TOTAL_QTY, "601");
+    verify(TestIdExtModules.LABEL_TOTALTOPAY, "90,303.60");
 
     tap(TestIdExtModules.BUTTON_PAY);
     verify(TestIdExtModules.LABEL_RECEIPT_TYPE, "Quotation - Under Evaluation");
@@ -62,8 +62,8 @@
     verify(TestIdExtModules.LABEL_POPUP_TITLE, "Not enough stock");
     tap(TestIdExtModules.BUTTON_POPUP_PROCEED);
     verify(TestIdExtModules.LABEL_RECEIPT_ROW1_QTY, "1");
-    verify(TestIdExtModules.LABEL_RECEIPT_ROW2_QTY, "100");
-    verify(TestIdExtModules.LABEL_RECEIPT_TOTAL_QTY, "101");
+    verify(TestIdExtModules.LABEL_RECEIPT_ROW2_QTY, "600");
+    verify(TestIdExtModules.LABEL_RECEIPT_TOTAL_QTY, "601");
 
     tap(TestIdExtModules.BUTTON_DELETERECEIPT);
     tap(TestIdExtModules.BUTTON_DELETERECEIPT_CONFIRM);
@@ -83,10 +83,10 @@
     verify(TestIdExtModules.LABEL_TOTALTOPAY, "154.10");
     // changes quantity
     tap(TestIdPack.BUTTON_RECEIPT_ROW2);
-    tap(TestIdPack.BUTTON_KEYPAD_1);
+    tap(TestIdPack.BUTTON_KEYPAD_6);
     tap(TestIdPack.BUTTON_KEYPAD_0);
     tap(TestIdPack.BUTTON_KEYPAD_0);
     tap(TestIdPack.BUTTON_KEYPAD_QUANTITY);
-    verify(TestIdExtModules.LABEL_TOTALTOPAY, "15,053.60");
+    verify(TestIdExtModules.LABEL_TOTALTOPAY, "90,303.60");
   }
 }
diff -r 51ce78428164 -r 27cf9832ab7a src-test/org/openbravo/test/mobile/retail/extmodules/selenium/tests/stockvalidation/I34092_OrderStockValidationMultiReceipt.java
--- a/src-test/org/openbravo/test/mobile/retail/extmodules/selenium/tests/stockvalidation/I34092_OrderStockValidationMultiReceipt.java	Thu Dec 21 17:02:55 2017 +0100
+++ b/src-test/org/openbravo/test/mobile/retail/extmodules/selenium/tests/stockvalidation/I34092_OrderStockValidationMultiReceipt.java	Fri Dec 22 12:00:20 2017 +0100
@@ -41,7 +41,7 @@
       enablePermission(true);
       // first ticket
       addProduct();
-      changeQuantity();
+      changeQuantity(false);
       verify(TestIdExtModules.LABEL_RECEIPT_ROW1_QTY, "50");
       verify(TestIdExtModules.LABEL_RECEIPT_TOTAL_QTY, "50");
       verify(TestIdExtModules.LABEL_TOTALTOPAY, "7,525.00");
@@ -49,7 +49,7 @@
       // second ticket
       tap(TestIdExtModules.BUTTON_NEWRECEIPT);
       addProduct();
-      changeQuantity();
+      changeQuantity(true);
       TestIdExtModules.CONFIRMATION_POPUP.enyoNode().waitUntilIsReady();
       verify(TestIdExtModules.LABEL_POPUP_TITLE, "Not enough stock");
       tap(TestIdExtModules.BUTTON_POPUP_CANCEL);
@@ -81,9 +81,15 @@
     verify(TestIdExtModules.LABEL_TOTALTOPAY, "150.50");
   }
 
-  private void changeQuantity() {
-    tap(TestIdPack.BUTTON_KEYPAD_5);
-    tap(TestIdPack.BUTTON_KEYPAD_0);
+  private void changeQuantity(boolean moreThanStocked) {
+    if (moreThanStocked) {
+      tap(TestIdPack.BUTTON_KEYPAD_6);
+      tap(TestIdPack.BUTTON_KEYPAD_0);
+      tap(TestIdPack.BUTTON_KEYPAD_0);
+    } else {
+      tap(TestIdPack.BUTTON_KEYPAD_5);
+      tap(TestIdPack.BUTTON_KEYPAD_0);
+    }
     tap(TestIdPack.BUTTON_KEYPAD_QUANTITY);
   }